src/App.tsx

```diff
@@ -4,64 +4,56 @@ import { ChannelEntry } from "./ChannelEntry";
 import { API_KEY } from "./BotAPI";
 import "./App.css";
 
-let channels: ChannelEntry[] = [];
+function Channels() {
+  let channelsArray: ChannelEntry[] = [];
 
-function populateChannelsArray(): void {
-  // Function responsible of correctly initialize above "channels" array
   for (let channelName of channelsNames) {
     let newChannel: ChannelEntry = {
       title: "",
-      link: "",
+      link: `https://t.me/${channelName}`,
       description: "",
       pictureID: "",
       subscribers: 0
     };
 
-    newChannel.title = channelName;
-    //getFormalName(channelName).then(r => newChannel.title = r);
-    newChannel.link = `https://t.me/${channelName}`;
-    //getPictureID(channelName).then(r => newChannel.pictureID = r);
+    getFormalName(channelName).then(r => newChannel.title = r);
+    getPictureID(channelName).then(r => newChannel.pictureID = r);
 
-    channels.push(newChannel);
+    Promise.all([getFormalName, getPictureID]).then(() => channelsArray.push(newChannel));
   }
 
-  function getFormalName(channelName: string): Promise<string> {
-    return fetch(`https://api.telegram.org/bot${API_KEY}/getChat?chat_id=@${channelName}`)
-      .then(r => r.json())
-      .then((r) => { return r.result.title; })
+  async function getFormalName(channelName: string): Promise<string> {
+    const r = await fetch(`https://api.telegram.org/bot${API_KEY}/getChat?chat_id=@${channelName}`);
+    const r_1 = await r.json();
+    return r_1.result.title;
   }
 
-  function getPictureID(channelName: string): Promise<string> {
-    return fetch(`https://api.telegram.org/bot${API_KEY}/getChat?chat_id=@${channelName}`)
-      .then(r => r.json())
-      .then((r) => { return r.result.photo.big_file_id; })
+  async function getPictureID(channelName: string): Promise<string> {
+    const r = await fetch(`https://api.telegram.org/bot${API_KEY}/getChat?chat_id=@${channelName}`);
+    const r_1 = await r.json();
+    return r_1.result.photo.big_file_id;
   }
-}
 
-function compare(a: ChannelEntry, b: ChannelEntry) {
-  if (a.subscribers < b.subscribers) return -1;
-  else if (a.subscribers > b.subscribers) return 1;
-  return 0;
-}
+  function compare(a: ChannelEntry, b: ChannelEntry) {
+    if (a.subscribers < b.subscribers) return -1;
+    else if (a.subscribers > b.subscribers) return 1;
+    return 0;
+  }
+  channelsArray.sort(compare);
+  console.log(channelsArray);
 
-function RenderChannels() {
-  channels.sort(compare);
   let key: number = 0;
-
   return (
     <div>
-      {channels.map((channel) => <ul className="channelsList" key={key++}>
+      {channelsArray.map((channel) => <ul className="channelsList" key={key++}>
```

Comment:
se utilizzi una variabile in un componente ReactJS che "cambia nel tempo" devi usare l'hook useState() qualora vuoi cambiare o usare il suo valore